src/heretic/model.py

```diff
@@ -49,15 +50,43 @@ def __init__(self, settings: Settings):
 
         self.model = None
 
-        for dtype in settings.dtypes:
-            print(f"* Trying dtype [bold]{dtype}[/]... ", end="")
+        # Filter dtypes: MPS doesn't support bfloat16, so skip "auto" on MPS
+        # (since "auto" typically resolves to bfloat16)
+        # Only apply this workaround on Intel x86_64 CPUs
+        
+        dtypes_to_try = settings.dtypes
+        use_mps = torch.backends.mps.is_available()
+        is_intel_x86_64 = platform.machine() == "x86_64"
+        if use_mps and is_intel_x86_64 and "auto" in dtypes_to_try:
+            dtypes_to_try = [dtype for dtype in dtypes_to_try if dtype != "auto"]
+            if not dtypes_to_try:
+                # If only "auto" was specified, default to float16 for MPS
+                dtypes_to_try = ["float16"]
+
+        for dtype in dtypes_to_try:
+            print(f"* Trying dtype [bold]{dtype_str}[/]... ", end="")
 
             try:
-                self.model = AutoModelForCausalLM.from_pretrained(
-                    settings.model,
-                    dtype=dtype,
-                    device_map=settings.device_map,
-                )
+                # For MPS on Intel x86_64, load to CPU first then convert dtype to avoid bfloat16 issues
+                if use_mps and is_intel_x86_64:
+                    # Load to CPU first without dtype to avoid bfloat16 preservation
+                    # Then convert dtype explicitly, then move to MPS
+                    self.model = AutoModelForCausalLM.from_pretrained(
```

Comment:
The same thing needs to happen in `reload_model` below, which is called on every trial.